x/circuit/keeper/keeper.go

```diff
@@ -80,3 +83,17 @@ func (k *Keeper) IsAllowed(ctx context.Context, msgURL string) (bool, error) {
 	has, err := k.DisableList.Has(ctx, msgURL)
 	return !has, err
 }
+
+func (k *Keeper) IsAllowedPreMessageHook(ctx context.Context, msg proto.Message) error {
+	_, err := k.IsAllowed(ctx, msgTypeURL(msg))
+	return err
+}
+
+// MsgTypeURL returns the TypeURL of a `sdk.Msg`.
+func msgTypeURL(msg proto.Message) string {
```

Comment:
Nit: We have an helper for in codec and types, that and we define this as well in core and server/v2. Maybe we should make the one of core public and use it everywhere instead.